#VRML_SIM R2023a utf8

IMPORTABLE EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/bitcraze/crazyflie/protos/Crazyflie.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"



WorldInfo {
}
Viewpoint {
  orientation -0.12577821568808165 0.8804431530844167 0.45716484406075336 0.8071583944978193
  position -0.6915100278560411 -0.36483795282963377 0.6236285609441987
  follow "Crazyflie"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0 0.015
  children [
    DEF BODY Transform {
      translation 0 0 -0.015
      children [
        CadShape {
          url [
            "../../meshes/collada_files/cf2_assembly.dae"
          ]
        }
      ]
    }
    DEF MULTIRANGER Transform {
      translation 0 0 0.02
      children [
        DistanceSensor {
          rotation 0 0 1 -1.57
          name "range_right"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 3.14
          name "range_back"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          rotation 0 0 1 1.57
          name "range_left"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
        DistanceSensor {
          name "range_front"
          lookupTable [
            0 0 0
            2 2000 0
          ]
        }
      ]
    }
    Camera {
      translation 0.03 0 0.01
      fieldOfView 0.87
      width 324
      height 324
    }
    GPS {
    }
    Gyro {
    }
    InertialUnit {
    }
    DEF M1 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 -0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m1_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation 0.031 -0.031 0.008
        children [
          DEF PROP_SHAPE Shape {
            appearance DEF PLASTIC PBRAppearance {
              baseColor 0 0 0
              metalness 0.3
            }
            geometry Mesh {
              url [
                "../../meshes/collada_files/ccw_prop.dae"
              ]
            }
          }
        ]
      }
    }
    DEF M2 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 -0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m2_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 -0.031 0.008
        children [
          USE PROP_SHAPE
        ]
      }
    }
    DEF M3 Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.031 0.031 0.008
      thrustConstants -4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m3_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix Solid {
        translation -0.031 0.031 0.008
        children [
          USE PROP_SHAPE
        ]
      }
    }
    DEF M4 Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.031 0.031 0.008
      thrustConstants 4e-05 0
      torqueConstants 2.4e-06 0
      device RotationalMotor {
        name "m4_motor"
        maxVelocity 600
        maxTorque 30
      }
      slowHelix DEF PROP Solid {
        translation 0.031 0.031 0.007
        children [
          USE PROP_SHAPE
        ]
      }
    }
  ]
  boundingObject Cylinder {
    height 0.03
    radius 0.05
  }
  physics Physics {
    density -1
    mass 0.05
  }
  controller "crazyflie_controller_py"
}
